log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileOutput.java

```diff
@@ -83,52 +89,80 @@ public boolean closeWriter() {
 
     /**
      * Writes a string to the end of the file
+     *
      * @param string String to append to file
-     * @exception IOException when the file cannot be accessed
+     * @throws IOException when the file cannot be accessed
      */
-    public void appendString( String string ) throws InvalidPathException, IOException {
+    public synchronized void appendString(String string) throws InvalidPathException, IOException {
```

Comment:
I'm thinking of putting the blocking queue at the Log_Engine level to avoid the pain further down but it does raise the issue of more messages coming in over the time it takes to write them to the file. As I understand it the problem is still going to be there though as multi-threaded writes will still need to sync up or we end up using random access writers which I'm not sure is such a good thing for a log file... Will need to think about that a bit more down the line.